homeassistant/components/tesla_wall_connector/config_flow.py

```diff
@@ -0,0 +1,163 @@
+            )
+            version = await wall_connector.async_get_version()
+        except WallConnectorError as ex:
+            _LOGGER.exception(
+                "Could not read serial number from Tesla WallConnector at [%s]: [%s]",
+                self.ip_address,
+                ex,
+            )
+            return self.async_abort(reason="cannot_connect")
+
+        self.serial_number = version.serial_number
+
+        await self.async_set_unique_id(self.serial_number)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: self.ip_address})
+
+        _LOGGER.info(
+            "No entry found for wall connector with IP %s. Serial nr: %s",
+            str(self.ip_address),
+            str(self.serial_number),
+        )
+
+        self.context["description_placeholders"] = {CONF_HOST: self.ip_address}
+        return await self.async_step_user()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        data_schema = vol.Schema(
+            {vol.Required(CONF_HOST, default=self.ip_address): str}
+        )
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=data_schema)
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except WallConnectorError:
+            errors["base"] = "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+
+        if not errors:
+            existing_entry = await self.async_set_unique_id(
+                info[WALLCONNECTOR_SERIAL_NUMBER]
+            )
+            if existing_entry:
+                self.hass.config_entries.async_update_entry(
+                    existing_entry, data=user_input
+                )
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="already_configured")
+
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for Tesla Wall Connector."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
```

Comment:
Do we need a listener setup in `async_setup_entry` to reload the config entry for this to take effect?
https://developers.home-assistant.io/docs/config_entries_options_flow_handler/#signal-updates

Reply:
While (or almost after) implementing the config update handler I realized that I could simply reload the integration by calling hass.config_entries.async_reload(entry.entry_id) instead of reading the new update interval from config and applying it the update coordinator. Since I decided to have two update intervals, one while charging and one while not charging, this code is not as simple as one would expect. I'm therefore wondering whether I should just drop that two intervals feature. What do you think?

Reply:
As long as the device can handle it, it would be ideal to make the code less complex and drop the second interval

Reply:
See 3714794

Next up: Improve test coverage